### PR TITLE
First VPN working version

### PR DESCRIPTION
**Pull request by jvinolas**
It will allow to connect directly to the guest ip using wireguard client vpn. Things missing:
- Filter which guests ip the client can access, based on the guest ip assigned. We could get those by one of this options:
  - a) Get the dnsmasq assigned ip by quering qemu-ga (guest agent) when the host is up (i don't like this one)
  - b) Get the dnsmasq assigned ip querying libvirt (we should wait for guest to get ip)
  - c) Assign fixed IPs to libvirt dnsmasq. Here we could pre-generate this config based on MAC or try to modify the dnsmasq config and reload lively? Or maybe just do some kind of virsh net-edit using libvirt
    `cat /var/lib/libvirt/dnsmasq/wireguard.conf 
strict-order
pid-file=/var/run/libvirt/network/wireguard.pid
except-interface=lo
bind-dynamic
interface=virbr20
dhcp-range=192.168.128.33,192.168.129.254,255.255.252.0
dhcp-no-override
dhcp-authoritative
dhcp-lease-max=478
dhcp-hostsfile=/var/lib/libvirt/dnsmasq/wireguard.hostsfile
addn-hosts=/var/lib/libvirt/dnsmasq/wireguard.addnhosts
dhcp-option=121,10.200.200.0/24,192.168.128.1
`
- Discuss where to put the vpn configuration to be easy for clients. It is weird that it shows the button in viewers modal for each desktop as it will be the same config for the user always. Also we need to discuss how to give a simple access button for RDP using VPN
- Create a Windows client vbscript that checks if wireguard is installed, if not it downloads and installs and then applies the downloaded user vpn config and starts RDP with it's configuration

### CODE REVIEW

File: webapp/webapp/webapp/static/js/viewer.js [hunk 1]
```diff
@@ -19,12 +19,18 @@ function setViewerButtons(id,socket,offer){
...
              'preferred': false
              },
-             {
+/*              {
              'type': 'spice',
              'client': 'websocket',
              'secure': true,
              'preferred': false
-             },
+             }, */
```
Comment (sim6) [inline webapp/webapp/webapp/static/js/viewer.js:27]: What about remove commented code?

File: docker/vpn/src/wgtools.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+#[Interface]
+#Address = 10.200.200.3/32
+#PrivateKey = CLIENT_PRIVATE_KEY
+#DNS = 10.200.200.1
+
+#[Peer]
+#PublicKey = SERVER_PUBLICKEY
+#Endpoint = my.ddns.example.com:51820
+#AllowedIPs = 192.168.128.0/22, ::/0
```
Comment (sim6) [inline docker/vpn/src/wgtools.py:225]: What about remove this examples?
```suggestion
```

**Review by sim6 — CHANGES_REQUESTED** [review]
What about use `black` on new python files.

File: docker-compose-parts/portal.yml [hunk 3]
```diff
@@ -3,13 +3,15 @@ services:
...
+    cap_add:
+    - NET_ADMIN    
     networks:
       - isard-network
     ports:
       - published: 80
         target: 80
       - published: 443
-        target: 443
+        target: 443
```
[the remark this answers is not in the document]
Comment (jvinolas) [reply docker-compose-parts/portal.yml:14]: I'll take a look at black formatter...

**Pull request closed without merging**